test/housekeeping_test.js

```diff
@@ -333,55 +334,48 @@ describe('House Keeper', () => {
     }
   });
 
-  it('should call sweepVolumes exactly once', async() => {
-    houseKeeperMock.expects("_sweepVolumes").exactly(regions.length);
+  it('should call handleVolumeData exactly once', async() => {
+    houseKeeperMock.expects("_handleVolumeData").once();
     
     await houseKeeper.sweep();
     houseKeeperMock.verify();
   });
 
-  it('should not fail if no volume data is returned', async() => {
-    houseKeeperMock.expects("_handleVolumeData").never();
-
+  it('should call describeVolumes endpoint exactly once per region if no NextToken is provided', async() => {
     await houseKeeper.sweep();
-    houseKeeperMock.verify();
+    assume(describeVolumesStub.callCount).equals(regions.length);
   });
-  
-  it('should call handleVolumeData exactly once per volume', async() => {
-    houseKeeperMock.expects("_handleVolumeData").twice();
-      
-    describeVolumesStub.withArgs(sinon.match(function(value) {
-      return value === ec2['us-west-2'] 
-    })).returns({
-       Volumes: [{
-         Attachments: [],
-         AvailabilityZone: 'us-west-2', 
-         CreateTime: new Date().toString(), 
-         Size: 8, 
-         SnapshotId: "snap-1234567890abcdef0", 
-         State: "in-use", 
-         VolumeId: "vol-049df61146c4d7901", 
-         VolumeType: "standard",
-       }]
-    });
 
+  it('should call describeVolumes endpoint again if NextToken is provided', async() => {
     describeVolumesStub.withArgs(sinon.match(function(value) {
-      return value === ec2['us-east-2']
-    })).returns({
+      return value === ec2['us-west-2'] 
+    })).onFirstCall().returns({
       Volumes: [{
-        Attachments: [], 
-        AvailabilityZone: 'us-east-2', 
+        Attachments: [],
+        AvailabilityZone: 'us-west-2', 
         CreateTime: new Date().toString(), 
-        Size: 16, 
-        SnapshotId: "snap-1234567890abcdef09", 
+        Size: 8, 
+        SnapshotId: "snap-1234567890abcdef0", 
         State: "in-use", 
-        VolumeId: "vol-049df61146c4d7902", 
+        VolumeId: "vol-049df61146c4d7901", 
         VolumeType: "standard",
-      }]
+      }],
+      NextToken: "1" 
+    }).onSecondCall().returns({
+      Volumes: [{
+        Attachments: [],
+        AvailabilityZone: 'us-west-2', 
+        CreateTime: new Date().toString(), 
+        Size: 8, 
+        SnapshotId: "snap-1234567890abcdef0", 
```

Comment:
nit: SnapshotId and VolumeId should be unique to each volume

Reply:
Yeah, weird. I fixed that in one spot but it seems to have gotten mixed up during the branching and merging..